crates/driver/src/extract_defrefs.rs

```diff
@@ -580,3 +619,71 @@ pub async fn extract_defrefs(
     write_json_file(extract.extracted_mappings.as_path(), &extracted_mappings)?;
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::fs;
+
+    async fn walk_directories<A, F, R>(
+        working_directory: &PathBuf,
+        child_path: String,
+        file_extensions: &Vec<OsString>,
+        test_globs: &Vec<String>,
+        source_config: SourceConfig,
+        on_entry: F,
+    ) -> Result<Vec<A>>
+    where
+        F: Fn(DirEntry, PathBuf) -> R,
+        R: futures::Future<Output = Result<A>> + Send + 'static,
+    {
+        let mut results: Vec<A> = Vec::default();
+        let globset = to_globset(test_globs)?;
+        let is_empty_globs = test_globs.is_empty();
+        for entry in WalkBuilder::new(working_directory.join(child_path))
```

Comment:
can we remove the duplication of this WalkBuilder filter? It's the same as the one above isn't it?

Reply:
Yea, so there's an identical block of code in `async fn async_extract_def_refs` and `walk_directories` (in test). I did originally implement `walk_directories` intending to split out and write tests for glob evaluation logic, but ran into ownership issue of accessing `opt: Arc<ExtractConfig>` from a lambda, which kept complaining that the lambda would outlive `opt`, so I've split out `has_good_extension` and `to_globset` so at least we have some tests. That was last year.

I can try again to convince the compiler that `opt` would live longer. Each time I try to make things testable by making function accept a lambda I run into this. Maybe I did `opt: &'static Opt,` last time?

Reply:
Addressed in dbf1e05.